### PR TITLE
docs: Update documentation structure with Platform as the first class citizen

### PR DESCRIPTION
**Pull request by kbarnard10**
### Background

Updating docs structure to reflect current state of project. 

### Changes 🏗️

- Adding new homepage content
- Restructuring navigation to move classic components to own section
- Renaming "docs" to "contribute" 
- Renaming "Setup" to "getting started" 
- Adding video embeds to setup guide and block creation 
- Updating favicon

### CODE REVIEW

**Comment by aarushik93** [comment]
Screenshot 2024-10-07 at 9 25 00 AM
The youtube link shows up like this for me

File: docs/content/classic/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- The [Forge](#forge)
+- The [Frontend](#frontend)
+
+To tie these together, we also have a [CLI] at the root of the project.
+
+## 🌐 Server
+
+<!-- Setup, then Advanced, then New Blocks -->
+
+**[📖 Setup](server/setup.md)**
```
Comment (ntindle) [inline docs/content/classic/index.md:19]: All of these references need updated since the file moved

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -95,14 +94,14 @@ theme:
...
-      scheme: default
+      scheme: dark
       toggle:
         icon: material/weather-night
         name: Switch to dark mode
 
     # Palette toggle for dark mode
     - media: "(prefers-color-scheme: dark)"
-      scheme: slate
+      scheme: light
```
Comment (Pwuts) [inline docs/mkdocs.yml:104]: I'm pretty sure this is the wrong way around

File: docs/content/server/getting-started.md [hunk 3]
```diff
@@ -1,9 +1,15 @@
-# Setting up the server
+# Getting Started with AutoGPT: Self-Hosting Guide
+
+This tutorial will walk you through the process of setting up AutoGPT locally on your machine.
+
+<center><iframe width="560" height="315" src="https://www.youtube.com/embed/4Bycr6_YAMI?si=dXGhFeWrCK2UkKgj" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></center>
+
+## Setting up the server
```
Comment (Pwuts) [inline docs/content/server/getting-started.md:7]: This is the only H2 header now. If you remove this, all the headers below can stay the same level/size.

File: docs/content/server/getting-started.md [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 
 - [Introduction](#introduction)
 - [Prerequisites](#prerequisites)
```
Comment (Pwuts) [inline docs/content/server/getting-started.md:10]: does this need to be expanded with "Cloning the Repository", "Running the backend services", etc?

File: docs/content/index.md [hunk 5]
```diff
@@ -2,126 +2,84 @@
...
+## Available Language Models
 
-- 🔌 **Agent Protocol Standardization** - AutoGPT uses the agent protocol from the AI Engineer Foundation to ensure compatibility with many agents, both from within and outside the project.
+The platform comes pre-integrated with cutting-edge LLMs:
 
----
+- OpenAI
+- Anthropic
+- Groq
+- Llama
```
Comment (Pwuts) [inline docs/content/index.md:70]: The first 3 are LLM *providers*, not LLMs. "Llama" is an LLM family, ranging from old and "primitive" to new and very intelligent.
```suggestion
- OpenAI (GPT-4 Turbo, GPT-4o, o1, etc.)
- Anthropic (Claude 3/3.5)
- Groq (Llama 3, Mixtral, Gemma)
- Ollama (Llama 3/3.1)
```

File: docs/content/classic/index.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+The AutoGPT project consists of four main components:
+
+- The [Server](#server) &ndash; known as the "AutoGPT Platform"
+- The [Agent](#agent) &ndash; also known as just "AutoGPT"
+- The [Benchmark](#benchmark) &ndash; AKA `agbenchmark`
+- The [Forge](#forge)
+- The [Frontend](#frontend)
+
+To tie these together, we also have a [CLI] at the root of the project.
```
Comment (Pwuts) [inline docs/content/classic/index.md:13]: Without context, this is confusing.
```suggestion
# AutoGPT Classic Documentation

Welcome to the AutoGPT Classic Documentation.

The classic AutoGPT project consists of four main components:

- The [Agent](#agent) &ndash; also known as just "AutoGPT"
- The [Benchmark](#benchmark) &ndash; AKA `agbenchmark`
- The [Forge](#forge)
- The [Frontend](#frontend)

To tie these together, we also have a [CLI] in `classic/`.
```

File: docs/content/classic/index.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+<!-- Setup, then Advanced, then New Blocks -->
+
+**[📖 Setup](server/setup.md)**
+&ensp;|&ensp;
+**[📖 Advanced Setup](server/advanced_setup.md)**
+&ensp;|&ensp;
+**[📖 Making New Blocks](server/new_blocks.md)**
+
+The server is the backbone of the New AutoGPT project. It provides the infrastructure for the agents to run, and the UI for you to interact with them. It integrates with the Forge, Agent, and a bespoke UI to provide a seamless experience.
```
Comment (Pwuts) [inline docs/content/classic/index.md:25]: Not part of the classic project -> shouldn't be in here

File: docs/content/index.md [hunk 8]
```diff
@@ -2,126 +2,84 @@
...
+-  **Deployment Controls**: Manage the lifecycle of your agents.
+-  **Ready-to-Use Agents**: Select from pre-configured agents.
+-  **Agent Interaction**: Run and interact with agents through a user-friendly interface.
+-  **Monitoring and Analytics**: Track agent performance and gain insights.
 
-- 💡 **Explore** - See what AI can do and be inspired by a glimpse of the future.
+##  Platform Components
 
-- 🚀 **Build with us** - We welcome any input, whether it's code or ideas for new features or improvements! Join us on [Discord](https://discord.gg/autogpt) and find out how you can join in on the action.
+### Agents and Workflows
```
Comment (Pwuts) [inline docs/content/index.md:41]: What are "workflows" here?

File: docs/content/index.md [hunk 9]
```diff
@@ -2,126 +2,84 @@
...
 
-The AutoGPT project consists of four main components:
+## What is the AutoGPT Platform?
 
-- The [Server](#server) &ndash; known as the "AutoGPT Platform"
-- The [Agent](#agent) &ndash; also known as just "AutoGPT"
-- The [Benchmark](#benchmark) &ndash; AKA `agbenchmark`
-- The [Forge](#forge)
-- The [Frontend](#frontend)
+The AutoGPT Platform is a groundbreaking system that revolutionizes AI utilization for businesses and individuals. It enables the creation, deployment, and management of continuous agents that work tirelessly on your behalf, bringing unprecedented efficiency and innovation to your workflows.
```
Comment (Pwuts) [inline docs/content/index.md:7]: "continuous agents" isn't defined further, ambiguous = confusing imo

File: docs/content/index.md [hunk 10]
```diff
@@ -2,126 +2,84 @@
...
-The heart of AutoGPT, and the project that kicked it all off: a semi-autonomous agent powered by LLMs to execute any task for you*.
+The user interface where you interact with the platform:
 
-We continue to develop this project with the goal of providing access to AI assistance to the masses, and building the future transparently and together.
+-  **Agent Builder**: Design and configure your own AI agents.
+-  **Workflow Management**: Build, modify, and optimize automation workflows.
+-  **Deployment Controls**: Manage the lifecycle of your agents.
+-  **Ready-to-Use Agents**: Select from pre-configured agents.
+-  **Agent Interaction**: Run and interact with agents through a user-friendly interface.
+-  **Monitoring and Analytics**: Track agent performance and gain insights.
```
Comment (Pwuts) [inline docs/content/index.md:37]: This division doesn't make a lot of sense to me. What's the goal with explaining the system's architecture here?

We have a front end and a back end, but the back end itself consists of 4 services, each with its own distinct role in the system.

File: docs/content/index.md [hunk 11]
```diff
@@ -2,126 +2,84 @@
...
-The server is the backbone of the New AutoGPT project. It provides the infrastructure for the agents to run, and the UI for you to interact with them. It integrates with the Forge, Agent, and a bespoke UI to provide a seamless experience.
+### 1. AutoGPT Server
 
----
+The powerhouse of our platform, containing:
 
-## 🤖 Agent
+-  **Source Code**: Core logic driving agents and automation processes.
+-  **Infrastructure**: Robust systems ensuring reliable and scalable performance.
+-  **Marketplace**: A comprehensive marketplace for pre-built agents.
```
Comment (Pwuts) [inline docs/content/index.md:26]: - Saying there is "source code" in our back end is a bit silly: the same can be said for the front end, and for everything else we've built.
- This is all very ambiguous and generic and doesn't say much about our actual back end
- "Marketplace" is a part of the system that has both front end and back end components

```suggestion
- **REST and WebSocket servers**: Services that act as our source-of-truth and expose APIs for other components like the front end.
- **Executor**: Executes agents asynchronously.
- **Scheduler**: Triggers the executor for scheduled runs.
```

**Pull request closed without merging**